### PR TITLE
Refactor and type four simple redux slices

### PR DESCRIPTION
**Pull request by eikhr**
# Description

Refactor redux slices `tags`, `forums`, `oauth2Applications` and `oauth2Grants` to use `legoAdapter` and `createSlice` instead of `createEntityReducer`.

# Result

Better type safety.

# Testing

- [x] I have thoroughly tested my changes.

Tested the relevant pages

---

ABA-688

### CODE REVIEW

File: app/routes/articles/components/ArticleList.tsx [hunk 1]
```diff
@@ -91,13 +91,10 @@ const ArticleList = () => {
...
+  usePreparedEffect('fetchPopularTags', () => dispatch(fetchPopular()), []);
   usePreparedEffect(
-    'fetchArticlesOverview',
-    () =>
-      Promise.allSettled([
-        dispatch(fetchPopular()),
-        dispatch(fetchAll({ next: false, query })),
-      ]),
+    'fetchArticleList',
+    () => dispatch(fetchAll({ next: false, query })),
```
Comment (ivarnakken) [inline app/routes/articles/components/ArticleList.tsx:97]: Why are these separated?

File: app/routes/tags/components/TagDetail.tsx [hunk 2]
```diff
@@ -31,9 +32,14 @@ const link = (key: string, tag: string) => {
...
+type TagDetailParams = {
+  tagId: string;
+};
 const TagDetail = () => {
-  const { tagId } = useParams();
-  const tag = useAppSelector((state) => selectTagById(state, { tagId }));
+  const { tagId } = useParams<TagDetailParams>() as TagDetailParams;
+  const tag = useAppSelector((state) =>
+    selectTagById(state, tagId),
+  ) as DetailedTag;
```
Comment (ivarnakken) [inline app/routes/tags/components/TagDetail.tsx:42]: Can't this be `undefined` as well? Or an empty object?

File: app/routes/users/components/UserSettingsOAuth2Form.tsx [hunk 3]
```diff
@@ -32,9 +32,9 @@ const UserSettingsOAuth2Form = () => {
   const { applicationId } = useParams<{ applicationId?: string }>();
   const isNew = applicationId === undefined;
   const application = useAppSelector((state) =>
-    selectOAuth2ApplicationById(state, {
-      applicationId,
-    }),
+    applicationId
+      ? selectOAuth2ApplicationById(state, applicationId)
+      : undefined,
```
Comment (ivarnakken) [inline app/routes/users/components/UserSettingsOAuth2Form.tsx:37]: Wouldn't it be "safer" and cleaner if this was just handled by `selectOAuth2ApplicationById`?

**Review by ivarnakken — APPROVED** [review]
Clean!

File: app/routes/articles/components/ArticleList.tsx (hunk 1 above)
Comment (eikhr) [reply app/routes/articles/components/ArticleList.tsx:97]: Wo don't need to re-fetch popular tags every time query changes

File: app/routes/tags/components/TagDetail.tsx (hunk 2 above)
Comment (eikhr) [reply app/routes/tags/components/TagDetail.tsx:42]: Yup, it can. I'll fix:)

File: app/routes/users/components/UserSettingsOAuth2Form.tsx (hunk 3 above)
Comment (eikhr) [reply app/routes/users/components/UserSettingsOAuth2Form.tsx:37]: yea, I suppose. But it's kind of pain to do that. The "ById"-selectors comes directly from redux-toolkits `entityAdapter`, and don't support selecting by undefined id.

Comment (eikhr) [reply app/routes/users/components/UserSettingsOAuth2Form.tsx:37]: It wasn't that bad, but it required some typecasting though. I added this and some more typing features in this PR: https://github.com/webkom/lego-webapp/pull/4604

**Pull request merged**